src/view/components/EditEntry/index.tsx

```diff
@@ -91,9 +92,9 @@ class EditEntry extends React.Component<PropsType, OwnStateType> {
   /**
    * Handle entry changes, update local state
    */
-  private onAmountChange = (newAmount: number) => this.setState(state => ({ amountOfProduct: newAmount }));
+  private onAmountChange = (newAmount: string) => this.setState(state => ({ amountOfProduct: parseInt(newAmount, radix) }));
   private onQualityChange = (newQuality: string) => this.setState(state => ({ quality: newQuality }));
-  private onRateChange = (newCostPerUnit: number) => this.setState(state => ({ costPerUnit: newCostPerUnit }));
+  private onRateChange = (newCostPerUnit: string) => this.setState(state => ({ costPerUnit: parseInt(newCostPerUnit, radix) }));
 
```

Comment:
I know this is super weird but this is what was causing everything to break. even though the type was supposed to be a number it was writing a string to the store which was breaking the selectors. Maybe onChangeText always passes a string even if you say you want it as a number?